lib/runtime/wasmer/exports.go

```diff
@@ -110,13 +113,18 @@ func (in *Instance) BabeGenerateKeyOwnershipProof(slot uint64, authorityID [32]b
 		return nil, fmt.Errorf("executing %s: %w", runtime.BabeAPIGenerateKeyOwnershipProof, err)
 	}
 
-	keyOwnershipProof := types.OpaqueKeyOwnershipProof{}
+	var keyOwnershipProof *types.OpaqueKeyOwnershipProof
 	err = scale.Unmarshal(encodedKeyOwnershipProof, &keyOwnershipProof)
 	if err != nil {
 		return nil, fmt.Errorf("scale decoding key ownership proof: %w", err)
 	}
 
-	return keyOwnershipProof, nil
+	// TODO check with team if returning error is preferred or should we keep just returning empty value
```

Comment:
I tend to agree with returning an error since we expect that key to report an equivocation

Reply:
I disagree. All these functions are only limited to just make a runtime call and return results as go structs.
If we get a nil key ownership proof, that still means that we called runtime function correctly and converted the go struct right way. This runtime func returns key ownership proof as scale encoded option. So, `keyOwnershipProof` could be nil.

I don't mind using `errEmptyKeyOwnershipProof`, but this is not the place where we use it.

Reply:
I think this is a good point, I will update